APIImpl/Network.lua

```diff
@@ -0,0 +1,64 @@
+-- Network.lua
+
+-- Provides an implementation for the network API. Unlike Cuberites API this is blocking.
+
+
+
+
+
+local socket = require 'socket'
+local http = require 'socket.http'
+local https = require 'ssl.https'
+local ltn12 = require("ltn12")
+
+
+
+
+
+local function MakeRequest(a_Method, a_Url)
+	local respbody = {}
+	local protocol = a_Url:match("^https") and https or http
+	local success, status, headers = protocol.request{
+		url = a_Url,
+		method = a_Method,
+		sink = ltn12.sink.table(respbody),
+	}
+	respbody = table.concat(respbody)
+	success = success ~= nil
+	if (success) then
+		status = nil
+	end
+	
+	return success ~= nil, status, respbody, headers
+end
+
+
+
+
+
+return {
+	["<static> cUrlClient:Delete(string, function)"] = function(a_Simulator, a_Self, a_Url, a_Callback)
+		local success, status, body, headers = MakeRequest("DELETE", a_Url)
+		a_Callback(body, headers)
```

Comment:
Don't call the callback directly, route it through the simulator:
```lua
a_Simulator:processCallbackRequest({
  Function = a_Callback,
  ParamValues = { body, headers },
  Notes = "cUrlClient:Delete request",
})
```
This way the simulator can check that callback in detail, too. And after #12 is implemented, it will even execute the callback after this one terminates, to improve detection.